Tests/U017.PSModules.AzureSearchFunctions.Tests.ps1

```diff
@@ -0,0 +1,62 @@
+Import-Module $PSScriptRoot\..\PSModules\AzureApiFunctions
+
+Describe "ApiRequest unit tests" -Tag "Unit" {
+
+    Mock -ModuleName AzureApiFunctions Invoke-WebRequest { 
+        return @{
+            StatusCode = 200
+            Content    = '{ "result": "Success" }'
+        }
+    }
+
+    $RequestParams = @{ 
+        Method = 'GET'
+        Url    = "https://api.mydomain.com/endpoint"
+        ApiKey = "Mock123"
+    }
+
+    It "Check ApiRequest passes the Content-Type abd api-key headers" {
+
+        ApiRequest @RequestParams
+
+        #$result.result | Should Be "Success"
+
+        Assert-MockCalled -ModuleName AzureApiFunctions Invoke-WebRequest -ParameterFilter {
+            $Headers['Content-Type'] -eq "application/json"
+        }
+        Assert-MockCalled -ModuleName AzureApiFunctions Invoke-WebRequest -ParameterFilter {
+            $Headers['api-key'] -eq "Mock123"
+        }
+
+    }
+
+    It "Check ApiRequest converts from JSON" {

```

Comment:
Will this test that it converts from JSON if you're not passing in a body

Reply:
Yes, the mock of invoke-webrequest returns the same value whether or not a body is passed in. This is basically testing the last line is converting the json text returned by the mock to an object

Reply:
got it